Sources/BSON/Double+BSONValue.swift

```diff
@@ -0,0 +1,22 @@
+import NIO
+
+extension Double: BSONValue {
+    static var bsonType: BSONType { .double }
+
+    var bson: BSON {
+        .double(self)
+    }
+
+    static func read(from buffer: inout ByteBuffer) throws -> BSON {
+        guard let data = buffer.readBytes(length: 8) else {
+            throw BSONError.InternalError(message: "Cannot read 8 bytes")
+        }
+        let value = data.withUnsafeBytes { $0.load(as: Double.self) }
```

Comment:
does this ever fail? / what happens if it can't read a double?

Reply:
I guess not, according to my research obtaining $0 which is a pointer to the underlying memory can't fail, we know its 8 bytes because of the readBytes call. The load is no more involved than a C-like cast, so at that point if the values are not in floating point format it will just be a garbled number.
Maybe I can call `Double(exactly: value)` and error check? Is that too much validation?

Reply:
From this [answer](https://stackoverflow.com/a/38024025) on stackoverflow, it looks like it's safer to copy the bytes to an already instantiated `Double` rather than loading one directly.